### PR TITLE
[WIP] Nicer more structured docs

### PR DESCRIPTION
**Pull request by robintibor**
Aim: 
Have more structure in the documentation so that people can more easily find all relevant documentation without going through the example list. 

Maybe also try whatever can be done to make the documentation more beautiful.

For now added some "Quickstart" example, will do more in coming days to improve.

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -41,6 +41,68 @@ alternatively, install the latest version of braindecode via pip:
...
+note the examples below for better-performing training pipelines.
+
+.. code-block:: python
+
+    import torch
+    from skorch.callbacks import LRScheduler
+    from skorch.helper import predefined_split
+    from braindecode.datasets.moabb import MOABBDataset
+    from braindecode.preprocessing.preprocess import preprocess, Preprocessor
+    from braindecode.preprocessing.windowers import create_windows_from_events
```
Comment (agramfort) [inline docs/index.rst:56]:
```suggestion
    from braindecode.preprocessing import preprocess, Preprocessor
    from braindecode.preprocessing import create_windows_from_events
```

File: docs/index.rst [hunk 2]
```diff
@@ -41,6 +41,68 @@ alternatively, install the latest version of braindecode via pip:
...
+    windows_dataset = create_windows_from_events(
+        dataset,
+        trial_start_offset_samples=0,
+        trial_stop_offset_samples=0,
+        preload=True,
+    )
+
+    # Create the model
+    model = ShallowFBCSPNet(
+        in_chans=train_set[0][0].shape[0],
```
Comment (agramfort) [inline docs/index.rst:80]: calling this n_channels seems so much more natural to me...

File: docs/index.rst [hunk 3]
```diff
@@ -41,6 +41,68 @@ alternatively, install the latest version of braindecode via pip:
...
+        Preprocessor('pick_types', eeg=True, meg=False, stim=False),  # Keep EEG sensors
+        Preprocessor(lambda x: x * 1e6) # Convert from V to uV
+    ]
+    preprocess(dataset, preprocessors)
+
+    # Cut trial windows from the data
+    windows_dataset = create_windows_from_events(
+        dataset,
+        trial_start_offset_samples=0,
+        trial_stop_offset_samples=0,
```
Comment (agramfort) [inline docs/index.rst:74]:
```suggestion
        start_sample=0,
        stop_sample=0,
```
would be neat :)

File: docs/index.rst [hunk 4]
```diff
@@ -41,6 +41,68 @@ alternatively, install the latest version of braindecode via pip:
...
+        preload=True,
+    )
+
+    # Create the model
+    model = ShallowFBCSPNet(
+        in_chans=train_set[0][0].shape[0],
+        n_classes=4,
+        input_window_samples=train_set[0][0].shape[1],
```
Comment (agramfort) [inline docs/index.rst:82]:
```suggestion
        n_samples=train_set[0][0].shape[1],
```
or n_times ...

File: examples/plot_bcic_iv_2a_moabb_cropped.py [hunk 5]
```diff
@@ -238,21 +232,20 @@
...
 # .. note::
-#    In this tutorial, we use some default parameters that we
-#    have found to work well for motor decoding, however we strongly
-#    encourage you to perform your own hyperparameter optimization using
-#    cross validation on your training data.
+#
+#     In this tutorial, we use some default parameters that we
+#     have found to work well for motor decoding, however we strongly
+#     encourage you to perform your own hyperparameter optimization using
+#     cross validation on your training data.
```
Comment (agramfort) [inline examples/plot_bcic_iv_2a_moabb_cropped.py:248]:
```suggestion
#     cross-validation on your training data.
```

**Review by agramfort — COMMENTED** [review]
some thoughts...

**Comment by bruAristimunha** [comment]
Hello @robintibor,

I am wondering if I could perhaps take advantage of your pull-request to change the site's theme from braindecode to pydata_sphinx_theme, such as numpy and mne. Would that be acceptable to you?

Probably, this involves a lot of small changes to make the site look nicer, and I would need some help.

Example: https://bruaristimunha.github.io/braindecode.github.io/master/index.html

**Comment by bruAristimunha** [comment]
many thanks @robintibor!

I created a PR in your folk.

**Comment by bruAristimunha** [comment]
For me there are still several details to review to fully integrate this new theme.  At mne it looks like this involved a team, we can just copy the the style from there if the license allows.  What do you think @agramfort?

**Comment by bruAristimunha** [comment]
Besides the details and adjustments in the div, I was wondering if the colours are a good start. 

In the light mode, it is not bad. I don't appreciated it much in the dark mode, and I suggest turning off this option. I was thinking about this palette in the light mode:

image

I don't know if someone will engage in this task with me, but I will try to centralize any decision here.

**Comment by agramfort** [comment]
I would encourage to have a look at how mne did its configuration.

regarding colors no strong feelings. Just make it with high enough contrast
for readability.

thx !

**Comment by bruAristimunha** [comment]
A first draft version of a logo:

As picture:
image

Vectorized version:
logo_braindecode

It's something to start with. Anyone want to give some feedback? @robintibor, @gemeinl, @agramfort

**Comment by agramfort** [comment]
how about something like this

Screenshot 2022-07-08 at 12 06 41

**Comment by bruAristimunha** [comment]
I liked Alexandre!

**Comment by sliwy** [comment]
How about putting the NN on the right inside the brain to match the left hemisphere?

something like this but with the brain from your options and EEG between the hemispheres:
image

in case, graphic from here: https://www.pngegg.com/en/png-bbroj

**Comment by bruAristimunha** [comment]
I like this option also @sliwy. The only thing I would add is a sign somewhere, but it could be near the name.

**Comment by bruAristimunha** [comment]
I like this option also @sliwy. The only thing I would add is a brain sign somewhere, but it could be near the name.

Let's wait for some maintainers and contributors to manifest, and we close with a vote. @hubertjb, @cedricrommel, @sylvchev, any contribution to the logo? 

I don't know the entire team involved in braindecode very well, so if you want to invite more people to encourage participation, please do.

**Comment by sliwy** [comment]
Primitive version but something like that:

image

and more compact one:

image

I would still work on the contrast of network graph inside the brain, maybe too similar to the brain color

**Comment by robintibor** [comment]
wow some really amazing options already. I think the network graph on top of the right brain looks a bit weird atm to me @sliwy Also partially sticking out etc. Otherwise also really nice

**Comment by sliwy** [comment]
how about this one? I would still increase the size of circles and line width. I can do more changes on Tuesday. @robintibor some details can be fixed, for now it is just few minutes in PowerPoint, finally we would need something better quality and vectorized.
image

and a little bit of fun python art (with missing eeg channel):
image (1)

**Comment by bruAristimunha** [comment]
I tried to incorporate all the suggestions and choices.

# Option 1- Emoji :eyes:
image

# Option 2 - Emoji :thumbsup:
image

# Option 3 - Emoji :grinning:
image

# Option 4 - Emoji :rocket:
image

# Option 5 - Emoji :rocket:
image

# Option 6 - Emoji :tada:
logo_braindecode


Verctorized version:
https://drive.google.com/file/d/1kI7M563oK2Wjx0XCjGPbFMpI7CXXQ6DE/view?usp=sharing

**Comment by agramfort** [comment]
I would go with option 6 but with NN not in black but in blue.

**Comment by robintibor** [comment]
What I like about option3/5's style of having braindecode on top is chance to make it more compact like sliwy's versions. But don't have too strong opinion on it, think many of these options do look very great already. I guess having a brain on the right may make it a bit confusing semantically. With brain only on left it is more clear: biological signal coming from brain, then processed by artificial neural network on right. Actually if one wants to have a neural network in brain, might make more sense to put in left brain and then have a neural network in anything machine-like on the right side...  but as said feel very happy with current options already also :)

**Comment by cedricrommel** [comment]
Both options 3 and 5 look good to me! Thank you for your investment on the
lib guys!! 🥲 Will be very nice to have a nice logo!!

Le dim. 10 juil. 2022 à 01:07, robintibor ***@***.***> a
écrit :

**Comment by bruAristimunha** [comment]
Look's like version 5 is the winner! I replication the mne configuration here, and change small things.

**Comment by agramfort** [comment]
I would make sure the width of the text of the logo matching the width of the visual

**Comment by bruAristimunha** [comment]
A thousand apologies for the mess, @agramfort. 

It looks like a test on augmentations is returning a RunTimeError, which is pretty weird because I didn't touch this area. While trying to find out the cause of the error, I will locally compile the rendered and put it on my fork.

**Comment by bruAristimunha** [comment]
I will rebase the nicer-doc branch with the current master branch, but the error does not seem related. 

In the meantime, I think I've made a first iteration of the new format for the site. I put it in my fork since I can't pass the CI. Link bellow:

https://bruaristimunha.github.io/braindecode.github.io/master/index.html

Can you look @agramfort and @robintibor?  I still have to correct the colours and aspect ratio of the images. So it would be nice to comment without being related to these two points.

**Comment by agramfort** [comment]
looks good !

I can you create a PR from current master branch to see if the CI failure
is specific to this branch?

**Comment by bruAristimunha** [comment]
Sure!

**Comment by bruAristimunha** [comment]
@agramfort and @gemeinl, can you take a look? (I think robin is on ICML)

https://output.circle-artifacts.com/output/job/34eeff2e-5be4-4956-9ae0-b42053e04f00/artifacts/0/dev/index.html

**Comment by agramfort** [comment]
besides a glitch in the footer LGTM

**Comment by bruAristimunha** [comment]
@agramfort, can you merge? 

To do this merge, we will have to change some things in the braindecode.github.io repository. I think there will be other issues with this documentation change, but I think we can fix it in other iterations.

**Comment by agramfort** [comment]
@bruAristimunha can you just fix the footer issue with the date?

**Comment by bruAristimunha** [comment]
Sure @agramfort. Done!

**Comment by agramfort** [comment]
Thx @bruAristimunha !

**Comment by bruAristimunha** [comment]
It's alive =) @robintibor and @agramfort https://braindecode.org/master/index.html#

It's going to be pretty cool when we reach version 0.7 =)

**Comment by robintibor** [comment]
Wow cool!
